lib/translator/common.cc

```diff
@@ -54,6 +54,22 @@ void SetAllocPageCallbacks(uint64_t (*alloc_callback)(uint16_t),
   dealloc_pages_callback = dealloc_callback;
 }
 
+StatusCode Allocation::SetPages(uint16_t data_page_count,
+                                uint16_t mdata_page_count) {
+  this->data_page_count = data_page_count;
+  this->data_addr = AllocPages(data_page_count);
+  this->mdata_page_count = mdata_page_count;
+  this->mdata_addr = AllocPages(mdata_page_count);
+
+  if (this->data_addr == 0 ||
+      (this->mdata_addr == 0 && mdata_page_count != 0)) {
+    DebugLog("Error when requesting memory");
+    return StatusCode::kFailure;
```

Comment:
Would it make sense to free the successful allocation, if only one failed?

Reply:
I believe there is a PR to call `FlushMemory` for anything that has been allocated even on failure, so I don't think we need to do any extra memory handling